common/djangoapps/student/management/commands/bulk_change_enrollment.py

```diff
@@ -92,10 +93,12 @@ def handle(self, *args, **options):
                 logger.info('Dry run, changes have not been saved. Run again with "commit" argument to save changes')
                 raise Exception('The --commit flag was not given; forcing rollback.')
 
-            # call `change_mode` which will change the mode and also emit tracking event
-            for enrollment in course_enrollments:
-                with transaction.atomic():
-                    enrollment.change_mode(mode=to_mode)
+            course_enrollments.update(mode=to_mode)
+
+            # Emit mode change event
+            updated_course_enrollments = CourseEnrollment.objects.filter(id__in=course_enrollments_ids)
```

Comment:
As we are filtering the queryset with `mode` as `CourseEnrollment.objects.filter(course_id=course_key, mode=from_mode)` and after, that queryset is updated with same mode field as `course_enrollments.update(mode=to_mode)`, I experienced update removes the pointer to the variable when we update the field used in filtering the queryset. I had to keep a copy of the IDs and re-fetch them. 

I also tried the following with no success and ended up with above solution. 
1. Assigning the queryset to a new variable.
2. Make copy with `copy` and `deepcopy`